### PR TITLE
Introduce telegrafreceiver

### PR DESCRIPTION
**Pull request by pmalek-sumo**
**Description:** Introduce telegraf receiver in unstable components set

This PR introduces telegrafreceiver that can use telegraf input plugins to ingest metrics and then forwards it to otc pipeline for processing.

An exemplar config that can be used for testing:

```yaml
receivers:
  telegraf:
    separate_field: false
    agent_config: |
      [agent]
        interval = "3s"
        flush_interval = "3s"
      [[inputs.mem]]
      [[inputs.disk]]

exporters:
  file:
    path: ./out.json

service:
  pipelines:
    metrics:
      receivers: [telegraf]
      exporters: [file]
```

### CODE REVIEW

File: receiver/telegrafreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Telegraf Receiver
```
Comment (pmm-sumo) [inline receiver/telegrafreceiver/README.md:1]: Maube a "limitations" section would be helfpul and a note there that currently only Gauges are supported?

Comment (pmalek-sumo) [reply receiver/telegrafreceiver/README.md:1]: Done, PTAL

**Comment by pmalek-sumo** [comment]
I gather that `loadtest` is destined to fail?

**Comment by pmm-sumo** [comment]
> I gather that `loadtest` is destined to fail?

In either case, it doesn't seem to be caused by us

**Review by pmm-sumo — APPROVED** [review]

**Comment by pmalek-sumo** [comment]
FYI: @mxiamxia @rakyll @alolita

**Comment by alolita** [comment]
Hi @pmm-sumo Can you please open this PR on the OpenTelemetry collector-contrib repo so that we can review? Ty!

**Comment by pmalek-sumo** [comment]
@alolita Created a PR with rebased version of this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2845.

Feel free to comment.

**Pull request merged**